src/legacy/core_plugins/timelion/server/routes/run.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/* eslint-disable @typescript-eslint/no-var-requires */
+import Joi from 'joi';
+import Bluebird from 'bluebird';
+import _ from 'lodash';
+import { Legacy } from 'kibana';
+// @ts-ignore
+import chainRunnerFn from '../handlers/chain_runner.js';
+const timelionDefaults = require('../lib/get_namespaced_settings')();
+
+export interface TimelionRequestQuery {
+  payload?: {
+    sheet: string[];
+    extended: {
+      es: {
+        filter: {
+          bool: {
+            filter: string[] | object;
+            must: string[];
+            should: string[];
+            must_not: string[];
+          };
+        };
+      };
+    };
+  };
+  query?: {
+    expression: string[];
+    from: string;
+    interval: string;
+    to: string;
+    timezone: string;
+  };
+}
+
+function formatErrorResponse(e: any, h: any) {
+  return h
+    .response({
+      title: e.toString(),
+      message: e.toString(),
+    })
+    .code(500);
+}
+
+const requestPayload = {
+  payload: Joi.object({
+    sheet: Joi.array().items(Joi.string()),
+    extended: Joi.object({
+      es: Joi.object({
+        filter: Joi.object({
+          bool: Joi.object({
+            filter: Joi.array().allow(null),
+            must: Joi.array().allow(null),
+            should: Joi.array().allow(null),
+            must_not: Joi.array().allow(null),
+          }),
+        }),
+      }),
+    }),
+    time: Joi.object({
+      from: Joi.string(),
+      interval: Joi.string(),
```

Comment:
I think `timezone` and `interval` can be marked `.required()` here, since I don't see that we can get into a state where they are not send.

Reply:
Done: https://github.com/elastic/kibana/pull/51788/files#diff-20042aff4267d58aaceb45fc0c1fb25bR87